yarn-project/node-lib/src/actions/snapshot-sync.ts

```diff
@@ -102,65 +95,111 @@ export async function trySnapshotSync(config: SnapshotSyncConfig, log: Logger) {
+      log.warn(
+        `Skipping snapshot from ${snapshotsUrl} as it has world state schema version ${snapshot.schemaVersions.worldState} but we expected ${WORLD_STATE_DB_VERSION}.`,
+        snapshot,
+      );
+      continue;
+    }
+
+    if (archiverL1BlockNumber && snapshot.l1BlockNumber < archiverL1BlockNumber) {
+      log.verbose(
+        `Skipping snapshot from ${snapshotsUrl} since local archiver is at L1 block ${archiverL1BlockNumber} which is further than snapshot at ${snapshot.l1BlockNumber}`,
+        { snapshot, archiverL1BlockNumber, snapshotsUrl },
+      );
+      continue;
+    }
+
+    if (archiverL1BlockNumber && snapshot.l1BlockNumber - Number(archiverL1BlockNumber) < minL1BlocksToTriggerReplace) {
+      log.verbose(
+        `Skipping snapshot from ${snapshotsUrl} as archiver is less than ${
+          snapshot.l1BlockNumber - Number(archiverL1BlockNumber)
+        } L1 blocks behind this snapshot.`,
+        { snapshot, archiverL1BlockNumber, snapshotsUrl },
+      );
+      continue;
+    }
+
+    snapshotCandidates.push({ snapshot, url: snapshotsUrl, fileStore });
   }
 
-  if (archiverL1BlockNumber && snapshot.l1BlockNumber < archiverL1BlockNumber) {
-    log.verbose(
-      `Skipping snapshot sync since local archiver is at L1 block ${archiverL1BlockNumber} which is further than last snapshot at ${snapshot.l1BlockNumber}`,
-      { snapshot, archiverL1BlockNumber },
-    );
+  if (snapshotCandidates.length === 0) {
+    log.verbose(`No valid snapshots found from any URL. Skipping snapshot sync.`, { ...indexMetadata, snapshotsUrls });
     return false;
   }
 
-  if (archiverL1BlockNumber && snapshot.l1BlockNumber - Number(archiverL1BlockNumber) < minL1BlocksToTriggerReplace) {
-    log.verbose(
-      `Skipping snapshot sync as archiver is less than ${
-        snapshot.l1BlockNumber - Number(archiverL1BlockNumber)
-      } L1 blocks behind latest snapshot.`,
-      { snapshot, archiverL1BlockNumber },
-    );
-    return false;
+  // Sort candidates by L1 block number (highest first)
+  snapshotCandidates.sort((a, b) => b.snapshot.l1BlockNumber - a.snapshot.l1BlockNumber);
+
+  // Try each candidate in order until one succeeds
+  for (const { snapshot, url } of snapshotCandidates) {
+    const { l1BlockNumber, l2BlockNumber } = snapshot;
+    log.info(`Attempting to sync from snapshot at L1 block ${l1BlockNumber} L2 block ${l2BlockNumber}`, {
+      snapshot,
+      snapshotsUrl: url,
+    });
+
+    try {
+      await snapshotSync(snapshot, log, {
+        dataDirectory: config.dataDirectory!,
+        rollupAddress: config.l1Contracts.rollupAddress,
+        snapshotsUrl: url,
+      });
+      log.info(`Snapshot synced to L1 block ${l1BlockNumber} L2 block ${l2BlockNumber}`, {
+        snapshot,
+        snapshotsUrl: url,
+      });
+      return true;
+    } catch (err) {
+      log.error(`Failed to download snapshot from ${url}. Trying next candidate.`, err, {
```

Comment:
Do we need to implement any cleanup logic here?

Reply:
That's handled on the `snapshotSync` function, which calls `tryRmDir` on the download dir in a `finally` block.